region/maridia/inner-green/Shaktool Room.json

```diff
@@ -646,6 +676,87 @@
         "Grapple teleporting crosses the room, so it will set the flag 'f_ShaktoolDoneDigging', assuming that the game is modified to treat the room symmetrically with how the flag gets set."
       ]
     },
+    {
+      "link": [2, 2],
+      "name": "R-Mode Entry",
+      "entranceCondition": {
+        "comeInWithRMode": {}
+      },
+      "requires": [],
+      "flashSuitChecked": true,
+      "blueSuitChecked": true,
+      "clearsObstacles": ["R-Mode"]
+    },
+    {
+      "link": [2, 2],
+      "name": "R-Mode Entry - Shaktool Done Digging",
+      "entranceCondition": {
+        "comeInWithRMode": {}
+      },
+      "requires": [
+        "f_ShaktoolDoneDigging"
+      ],
+      "flashSuitChecked": true,
+      "blueSuitChecked": true,
+      "clearsObstacles": ["A", "B", "R-Mode"],
+      "devNote": "Entering the room with the flag 'f_ShaktoolDoneDigging' set means that the sand blocks and Power Bomb blocks will not spawn."
+    },
+    {
+      "link": [2, 2],
+      "name": "R-Mode Spark Interrupt (Gain Blue Suit)",
+      "requires": [
+        {"obstaclesCleared": ["R-Mode", "B"]},
+        {"or": [
+          "h_CrystalFlashForReserveEnergy",
+          {"and": [
+            "h_RModeCanRefillReserves",
+            {"or": [
+              {"and": [
+                {"obstaclesCleared": ["A"]},
+                {"or": [
+                  "canWaterShineCharge",
+                  "Gravity"
+                ]},
+                "h_shinechargeMaxRunway",
+                {"shinespark": {"frames": 4, "excessFrames": 0}},
+                {"or": [
+                  {"and": [
+                    {"resourceMissingAtMost": [{"type": "Missile", "count": 0}]},
+                    {"partialRefill": {"type": "ReserveEnergy", "limit": 60}}
+                  ]},
+                  {"partialRefill": {"type": "ReserveEnergy", "limit": 20}}
+                ]}
+              ]},
+              {"and": [
+                "canBePatient",
```

Comment:
Is canBePatient on clearing the sand or the 1/3 on big energy?

Reply:
It's also worth noting that you'll get a small or big energy 2/3 of the time, and then you can get blue and kill the second snail before leaving to get another guaranteed energy drop (50/50 small/big).

So you'd have a 1/3 chance of failure (no drop on the first snail)
a 1/6 chance of ending with just 9 energy (small, small)
and a 1/2 chance of ending with 20+

Reply:
I wonder if we should have a `canComplexRModeSparkInterrupt`, which could mention nontrivial farming and a chance for bad drops instead of patience. 

I'm fine if patience is removed here or if it's kept with a fixme to put in a different tech. I don't think it really makes sense to have an OR for 10 vs 20 E remaining, as that's not going to change much, but I am also fine either way.

Reply:
It helps that, for example in a MapRando context, `canBePatient` keeps these less reliable strats out of Expert.

If it can do the same (or even if it doesn't) `canRelyOnLuck` would be perfect for this role in R-Mode Sparks and make it more clear that the issue is one of randomness, rather than simply taking a long time.